contrib/internal/httptrace/inferred_proxy.go

```diff
@@ -0,0 +1,138 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016 Datadog, Inc.
+
+package httptrace
+
+import (
+	"fmt"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/ext"
+	"net/http"
+	"strconv"
+	"time"
+
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"
+	"gopkg.in/DataDog/dd-trace-go.v1/internal/globalconfig"
+	"gopkg.in/DataDog/dd-trace-go.v1/internal/log"
+)
+
+type StartSpanOption = ddtrace.StartSpanOption
+
+const (
+	ProxyHeaderSystem      = "X-Dd-Proxy"
+	ProxyHeaderStartTimeMs = "X-Dd-Proxy-Request-Time-Ms"
+	ProxyHeaderPath        = "X-Dd-Proxy-Path"
+	ProxyHeaderHttpMethod  = "X-Dd-Proxy-Httpmethod"
+	ProxyHeaderDomain      = "X-Dd-Proxy-Domain-Name"
+	ProxyHeaderStage       = "X-Dd-Proxy-Stage"
+)
+
+type ProxyDetails struct {
+	SpanName  string `json:"spanName"`
+	Component string `json:"component"`
+}
```

Comment:
```suggestion
type proxyDetails struct {
	spanName  string `json:"spanName"`
	component string `json:"component"`
}
```

this type is not used outside of this package so it can be unexported